### PR TITLE
added gh api call for fetching llvm version

### LINKED ISSUE #1877: Speed up by querying Github API
https://github.com/bytecodealliance/wasm-micro-runtime/blob/b92075e283bd574ea8b809845031f86b1879c67a/.github/workflows/build_llvm_libraries.yml#L29-L32

Instead of cloning llvm-project repo, use [a Github API](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits) to list the latest commit.

### PR DESCRIPTION
**Pull request by hritikgupta**
Fixes #1877

### CODE REVIEW

File: build-scripts/build_llvm.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import argparse
 import os
 import pathlib
+import requests
```
Comment (loganek) [inline build-scripts/build_llvm.py:10]: By adding this we probably also need to add `requirements.txt` as this is not a built in module.

File: build-scripts/build_llvm.py [hunk 2]
```diff
@@ -232,14 +244,17 @@ def main():
         "arc": {
             "repo": "https://github.com/llvm/llvm-project.git",
             "branch": "release/15.x",
+            "owner_project": "llvm/llvm-project",
```
Comment (loganek) [inline build-scripts/build_llvm.py:247]: do we need to replicate that information? It's already available in the `repo` field.

File: .github/workflows/build_llvm_libraries.yml [hunk 3]
```diff
@@ -28,7 +28,7 @@ jobs:
```
Comment (lum1n0us) [inline .github/workflows/build_llvm_libraries.yml:28]: may need to install python dependencies(from requirements.txt) before

File: build-scripts/build_llvm.py (hunk 1 above)
Comment (lum1n0us) [reply build-scripts/build_llvm.py:10]: @loganek shall we create a global requirements.txt under */repo* or just create a local requirements.txt for *build_llvm.py* only.

File: build-scripts/build_llvm.py [hunk 4]
```diff
@@ -27,11 +28,22 @@ def clone_llvm(dst_dir, llvm_repo, llvm_branch):
...
 
 
-def query_llvm_version(llvm_dir):
-    GIT_LOG_CMD = f"git log --format=format:'%h' -1"
-    return subprocess.check_output(
-        shlex.split(GIT_LOG_CMD), cwd=llvm_dir, universal_newlines=True, text=True
-    )
+def query_llvm_version(llvm_info):
+    github_token = os.environ['GH_TOKEN']
+    url = f"https://api.github.com/repos/{llvm_info['owner_project']}/commits/{llvm_info['branch']}"
```
Comment (lum1n0us) [inline build-scripts/build_llvm.py:33]: just a minor one: `owner_project` -> `ownwer`

File: build-scripts/build_llvm.py (hunk 2 above)
Comment (hritikgupta) [reply build-scripts/build_llvm.py]: done

File: build-scripts/build_llvm.py (hunk 4 above)
Comment (hritikgupta) [reply build-scripts/build_llvm.py]: the string is a combination of owner and project, isn't it?

File: build-scripts/build_llvm.py (hunk 1 above)
Comment (hritikgupta) [reply build-scripts/build_llvm.py:10]: done, but i've added it locally under build-scripts

Comment (loganek) [reply build-scripts/build_llvm.py:10]: I think we can keep the file local; looks like that's what we do for `wasi-nn` tests and `language-bindings` for python as well.

File: build-scripts/requirements.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+requests
```
Comment (loganek) [inline build-scripts/requirements.txt:1]: no minimum version?

File: build-scripts/build_llvm.py (hunk 1 above)
Comment (hritikgupta) [reply build-scripts/build_llvm.py:10]: 👍 , kept it local for the same reason

File: build-scripts/requirements.txt (hunk 5 above)
Comment (hritikgupta) [reply build-scripts/requirements.txt]: added latest

**Review by loganek — COMMENTED** [review]
LGTM

**Review by loganek — APPROVED** [review]

**Pull request merged**